### PR TITLE
Makefile.include: pass IOTLAB_NODE to docker

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

Release testing I cam upon this corner case, not sure this is the best fix, open to ideas.

By default if IoT-LAB Cli Tools V3 is used then BINFILE is used to
flash on IoT-LAB. But BINFILE is not built by default when RIOT_CI_BUILD
is set as a ci optimization.

But since before IOTLAB_NODE was not passed to docker when building it
did not know that it should BUILD BINFILE as well, which led to failures
if doing:

$ IOTLAB_NODE=iotlab-m3.grenoble.iot-lab.info BOARD=iotlab-m3 \
  RIOT_CI_BUILD=1 BUILD_IN_DOCKER=1 make -C examples/hello-world/ flash

But if IOTLAB_NODE is passed at is checks for IoT-LAB cli Tools also
happen in the docker container which leads to a make error since those
are not present in docker.

Therefore avoid including the IoT-LAB makefiles when in docker, but add
BINFILE to BUILD_FILES if IOTLAB_NODE is set

### Testing procedure

$ IOTLAB_NODE=iotlab-m3.103.grenoble.iot-lab.info BOARD=iotlab-m3 \
  RIOT_CI_BUILD=1 BUILD_IN_DOCKER=1 make -C examples/hello-world/ flash

Fails in master succeeds with this PR

### CODE REVIEW

**Comment by fjmolinas** [comment]
@aabadie

**Comment by aabadie** [comment]
In the `test-on-iotlab` GH action, only adapting `BUILD_FILES` is needed:

https://github.com/RIOT-OS/RIOT/blob/b985a7489429bfc85ff45ac7d68a509f94ff9213/.github/workflows/test-on-iotlab.yml#L73-L76

I don't know how but `IOTLAB_NODE` is set correctly even when using `BUILD_IN_DOCKER` (or this action is also broken).

**Comment by aabadie** [comment]
> I don't know how but IOTLAB_NODE is set correctly even when using BUILD_IN_DOCKER

I know: `IOTLAB_NODE` doesn't need to be known from Docker, since it's only used by the flasher.

File: Makefile.include [hunk 1]
```diff
@@ -385,6 +385,7 @@ export PREFIX ?= $(if $(TARGET_ARCH),$(TARGET_ARCH)-)
 ifneq (,$(IOTLAB_NODE))
   PROGRAMMER ?= iotlab
   # iotlab uses ELFFILE by default for flashing boards.
+  BUILD_FILES += $(BINFILE)
```
Comment (aabadie) [inline Makefile.include:388]: I would only do this is `RIOT_CI_BUILD` is set:

```suggestion
  ifeq (1,$(RIOT_CI_BUILD))
    BUILD_FILES += $(BINFILE)
  endif
```

File: makefiles/docker.inc.mk [hunk 2]
```diff
@@ -67,6 +67,7 @@ export DOCKER_ENV_VARS += \
   SIZE \
   TOOLCHAIN \
   TEST_KCONFIG \
+  IOTLAB_NODE \
```
Comment (aabadie) [inline makefiles/docker.inc.mk:70]: Can we keep the alphabetical order (which as I saw it is already a bit broken...) ?

**Comment by aabadie** [comment]
I came up with a similar approach (not sure it's better):

```diff
diff --git a/Makefile.include b/Makefile.include
index a91c125ecc..316f8b3509 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -386,6 +386,17 @@ ifneq (,$(IOTLAB_NODE))
   PROGRAMMER ?= iotlab
   # iotlab uses ELFFILE by default for flashing boards.
   FLASHFILE ?= $(ELFFILE)
+  # RIOT_CI_BUILD disable the build of BINFILE which is required for flashing
+  # on IoT-LAB
+  ifeq (1,$(RIOT_CI_BUILD))
+    BUILD_FILES += $(BINFILE)
+  endif
+  # Disable IOTLAB_NODE is inside Docker to avoid including the
+  # iotlab.single.inc.mk file which is useless there: it's only useful for
+  # flashing and this is done outside of Docker.
+  ifeq ($(INSIDE_DOCKER),1)
+    IOTLAB_NODE :=
+  endif
 endif
 
 # Add standard include directories
diff --git a/makefiles/docker.inc.mk b/makefiles/docker.inc.mk
index cfa6cc7c4e..c32a244993 100644
--- a/makefiles/docker.inc.mk
+++ b/makefiles/docker.inc.mk
@@ -47,6 +47,7 @@ export DOCKER_ENV_VARS += \
   ELFFILE \
   HEXFILE \
   FLASHFILE \
+  IOTLAB_NODE \
   LINK \
   LINKFLAGPREFIX \
   LINKFLAGS \
```

The `test-on-iotlab` GH action has also to be updated.

**Comment by fjmolinas** [comment]
Applied suggested changes

**Review by aabadie — APPROVED** [review]
Running the `test-on-iotlab` action [on my fork](https://github.com/aabadie/RIOT/runs/4967207263?check_suite_focus=true) and it works ok.

Code changes are good.

ACK

**Pull request merged**